### PR TITLE
Add out of the box logging

### PR DESCRIPTION
**Pull request by enriquejosepadilla**
Source Generator: 
* Added generalized generate files for common code.
* Added generate admin and templates code. (Templates_templates are templates for django (html))

Django Admin Files:
* Admin code that shows previous 10 log lines from the project. Found at admin/cloud
* Does not show logs locally.
* Added Admin Html that shows a list of logs.

Service Accounts:
* Added logging service account with logging/read level

Config Changes:
* Installed apps replaces default admin for custom one.
* Yaml file allows to read secrets from the logging service.

### CODE REVIEW

File: django_cloud_deploy/skeleton/source_generator.py [hunk 1]
```diff
@@ -128,38 +127,70 @@ def generate_project_files(self,
...
-                           destination,
-                           template_folder=APP_TEMPLATE_FOLDER):
+    def generate_app_files(self, app_name, destination):
         """Create a django app using our template.
 
         Args:
-            app_name: str, name of the app to be created.
-            destination: str, the destination path to hold files of the app.
-            template_folder: str, the template folder name
+          app_name: str, name of the app to be created.
```
Comment (wangxf123456) [inline django_cloud_deploy/skeleton/source_generator.py]: 4 spaces Indentation

File: django_cloud_deploy/workflow/data/service_accounts.json [hunk 2]
```diff
@@ -4,9 +4,17 @@
...
-            "roles/cloudsql.client",
-            "roles/cloudsql.editor",
-            "roles/cloudsql.admin"
+          "roles/cloudsql.client",
+          "roles/cloudsql.editor",
+          "roles/cloudsql.admin"
+        ]
+    },
+    {
+        "id": "logging-credentials",
```
Comment (wangxf123456) [inline django_cloud_deploy/workflow/data/service_accounts.json]: I did not see modifications of the yaml file. There should be another secret.

File: django_cloud_deploy/skeleton/templates/admin_template/admin.py-tpl [hunk 3]
```diff
@@ -1 +1,69 @@
-#Custom Admin Code
\ No newline at end of file
+import os
+
+from django.contrib import admin
+from django.template.response import TemplateResponse
+from django.urls import path
+from google.cloud import logging
+from google.oauth2 import service_account
+
```
Comment (wangxf123456) [inline django_cloud_deploy/skeleton/templates/admin_template/admin.py-tpl]: Use 2 newlines.

File: django_cloud_deploy/skeleton/templates/admin_template/admin.py-tpl [hunk 4]
```diff
@@ -1 +1,69 @@
-#Custom Admin Code
\ No newline at end of file
+import os
```
Comment (wangxf123456) [inline django_cloud_deploy/skeleton/templates/admin_template/admin.py-tpl]: We should use 4 spaces indentation for this file to be consistent with other Django files.

File: django_cloud_deploy/skeleton/templates/admin_template/admin.py-tpl [hunk 5]
```diff
@@ -1 +1,69 @@
...
+  def cloud_view(self, request):
+    logs = self.get_logs()
+    context = dict(
+        # Include common variables for rendering the admin template.
+        self.each_context(request),
+        # Anything else you want in the context...
+        logs=logs
+    )
+    return TemplateResponse(request,
+                            "cloud_admin/cloud.html",
```
Comment (wangxf123456) [inline django_cloud_deploy/skeleton/templates/admin_template/admin.py-tpl]: Single quote to be consistent.

File: django_cloud_deploy/skeleton/templates/admin_template/admin.py-tpl [hunk 6]
```diff
@@ -1 +1,69 @@
...
+    query_filter = (
+      ['resource.type="container"',
+       'resource.labels.cluster_name="{{ project_name }}"',
+       'resource.labels.namespace_id="default"',
+       'resource.labels.project_id="{{ project_id }}"',
+       'resource.labels.zone:"us-west1-a"',
+       'resource.labels.container_name="{{ project_name }}-app"'])
+
+    query_filter = ' AND '.join(query_filter)
+    from google.cloud.logging import DESCENDING
```
Comment (wangxf123456) [inline django_cloud_deploy/skeleton/templates/admin_template/admin.py-tpl]: Is it fine to put this import to the top?

File: django_cloud_deploy/skeleton/source_generator.py [hunk 7]
```diff
@@ -128,38 +127,70 @@ def generate_project_files(self,
...
+
+    def generate_templates_files(self, destination):
+        """Create the django templates using our template.
+
+        Args:
+          destination: str, the destination path to hold files.
+        """
+        templates_destination = os.path.join(destination, 'templates')
+        self._generate_files(self.TEMPLATES_TEMPLATE_FOLDER,
+                             templates_destination, )
```
Comment (wangxf123456) [inline django_cloud_deploy/skeleton/source_generator.py]: Looks like something is missing?

File: django_cloud_deploy/skeleton/templates/admin_template/admin.py-tpl [hunk 8]
```diff
@@ -1 +1,69 @@
...
+
+    # Show ten entries
```
Comment (wangxf123456) [inline django_cloud_deploy/skeleton/templates/admin_template/admin.py-tpl]: Will we show more than 10 logs in the future?

Comment (enriquejosepadilla) [reply django_cloud_deploy/skeleton/templates/admin_template/admin.py-tpl]: maybe, from what I understand this is more of a POC regarding having more integration with GCP.

File: django_cloud_deploy/skeleton/source_generator.py [hunk 9]
```diff
@@ -178,31 +181,31 @@ def generate_admin_files(self, project_id, project_name, destination):
...
+                             templates_destination)
 
     def _generate_files(self, folder_name: str, destination: str,
                         filename_template_replacement=None, options=None):
         """Consolidates duplicate code that calls render_directory.
 
         Args:
-          folder_name: str, name of the folder that holds the templates
-          destination: str, the destination path to hold files of the app.
+            folder_name: str, name of the folder that holds the templates
```
Comment (wangxf123456) [inline django_cloud_deploy/skeleton/source_generator.py]: We can also delete types in documentation here.

File: django_cloud_deploy/skeleton/source_generator.py [hunk 10]
```diff
@@ -117,46 +119,77 @@ def generate_project_files(self, site: str, destination: str):
...
+        self._generate_files(self.TEMPLATES_TEMPLATE_FOLDER,
+                             templates_destination)
+
+    def _generate_files(self, folder_name: str, destination: str,
+                        filename_template_replacement=None, options=None):
+        """Consolidates duplicate code that calls render_directory.
+
+        Args:
+            folder_name: Name of the folder that holds the templates
+            destination: The destination path to hold files of the app.
```
Comment (wangxf123456) [inline django_cloud_deploy/skeleton/source_generator.py]: documentation of all parameters.

**Pull request merged**